test/spmd/test_xla_sharding.py

```diff
@@ -223,10 +223,13 @@ def test_mark_sharding_4d(self):
 
   def test_mark_sharding_partial(self):
     device = xm.xla_device()
-    t1 = torch.randn(4, 4).to(xm.xla_device())
-    t2 = torch.randn(4, 4).to(xm.xla_device())
-    expected = (t1 @ t2).cpu()
+    t1 = torch.randn(4, 4)
+    t2 = torch.randn(4, 4)
+    # Somehow the eager cpu result is different from the xla result.
+    expected = (t1.to(device) @ t2.to(device)).cpu()
```

Comment:
why do we need to make changes here?

Reply:
Because the xtensors used in expected should be copies of the ones used in actual spmd. The previous tests doesn't take care of those and t1 after expected calculation becomes IR.

Reply:
ok, can we do 
```
    t1 = torch.randn(4, 4).to(xm.xla_device())
    t2 = torch.randn(4, 4).to(xm.xla_device())
    expected = (t1 @ t2).cpu()
    xm.mark_step()
```
instead? This way we don't need to move same tensor to xla device multiple times. You can leave a comment explaining why mark_step is needed.

Reply:
Let me try. That should work too.

Reply:
Fixed.